corehq/apps/domain/decorators.py

```diff
@@ -225,6 +234,30 @@ def _inner(request, *args, **kwargs):
     return _inner
 
 
+def check_lockout(fn):
+    @wraps(fn)
+    def _inner(request, *args, **kwargs):
+        username = _get_username_from_request(request)
+        user = CouchUser.get_by_username(username)
```

Comment:
it would be nice if we didn't have to get the user from the DB twice in the non-lockout case (here and inside `_inner` above), but i guess this is only called when you aren't authenticated so probably not a big deal